### ISSUE TITLE
Type packages describing ESM do not work with `moduleResolution: "Node16"`

### ISSUE THREAD

**Issue by tpluscode** [DefinitelyTyped/DefinitelyTyped#64856]
Please see the issue https://github.com/microsoft/TypeScript/issues/53349

tl;dr; weird things happen when a types package exporting modules is used in a project where `moduleResolution` is set to `Node16` or `NodeNext`

This has come up multiple times already on DT:

https://github.com/DefinitelyTyped/DefinitelyTyped/pull/64331
https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/63363
https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/64148
https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/63096
https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/63348

The last one is a bit different since yargs is a dual package, but it has the property `type: module` all the same.

Should DT add some lint rule to ensure that type which describe ESM packages have `type: module` in their `package.json` too?

Or even better, maybe that field should be automatically set to match the upstream package?

**Comment by tpluscode** [comment]
Aha, I see this mentioned in the readme

> If the implementation package uses ESM and specifies "type": "module", then you should add a package.json with the same:

This is possibly no longer accurate. `"type": "module"` appears to have become a necessity. One must add it and not merely should

**Comment by sandersn** [comment]
This is probably relevant: https://github.com/microsoft/definitelytyped-tools/pull/647

**Comment by andrewbranch** [comment]
> One must add it and not merely should

Yeah, ideally this should be checked in CI, along with `exports` structure.

**Comment by sandersn** [comment]
@jakebailey and I are working on this now, but the first step is to switch to a modern monorepo structure and add correct package.jsons for every package on DT.

**Comment by tpluscode** [comment]
Did I get that right, the DT repo will be a monorepo? Won't the size be a problem 😵‍💫

**Comment by jakebailey** [comment]
No, if you want the entire thing installed, it takes less than 2 minutes, and you can still do a partial install to work on a subset of packages and go faster.

**Comment by samestep** [comment]
Ran into this today when trying to use `yargs` and `@types/yargs` in TypeScript; is there a workaround?

**Comment by jakebailey** [comment]
dtslint now runs AreTheTypesWrong, so this should be fixed.
